### PR TITLE
fix(vaadin-grid): Material Grid Improvements

### PR DESCRIPTION
**Pull request by platosha**
Fixes vaadin/vaadin-grid#807

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-material-theme/45)

### CODE REVIEW

**Comment by jouni** [comment]
screen shot 2018-05-21 at 9 25 45

When you resize the columns as small as possible, the contents overflow the cells.

File: vaadin-grid-sorter.html [hunk 1]
```diff
@@ -0,0 +1,71 @@
+<link rel="import" href="color.html">
+<link rel="import" href="font-icons.html">
+<link rel="import" href="typography.html">
```
Comment (jouni) [inline vaadin-grid-sorter.html]: Unnecessary import.

File: test/all-theme-imports.html [hunk 2]
```diff
@@ -10,6 +10,12 @@
 <link rel="import" href="../vaadin-date-picker.html">
 <link rel="import" href="../vaadin-dialog.html">
 <link rel="import" href="../vaadin-dropdown-menu.html">
+<link rel="import" href="../vaadin-grid-column.html">
```
Comment (jouni) [inline test/all-theme-imports.html]: No import for `vaadin-grid.html`?

**Comment by jouni** [comment]
screen shot 2018-05-21 at 9 34 09

Is the selection color applied for the detail row on purpose? See https://github.com/vaadin/vaadin-grid/issues/1197 for reference.

File: vaadin-grid-sorter.html (hunk 1 above)
Comment (platosha) [reply vaadin-grid-sorter.html]: Good catch, removed.

File: test/all-theme-imports.html (hunk 2 above)
Comment (platosha) [reply test/all-theme-imports.html]: It’s there below. On my system, `ls` does sort the files this way. I can’t notice any errors with that order, so I’d keep it as it is.

**Comment by platosha** [comment]
>  Do we want to introduce the dense variant already? I removed it from the button, and was planning on doing the same for text field. I think it might make sense to have a more solid concept around the density, similar to what we have in Lumo with sizing & spacing.

Yeah maybe not. Removed that. It’s a very neat feature for the grid, however.

> When you resize the columns as small as possible, the contents overflow the cells.

Fixed.

> Is the selection color applied for the detail row on purpose? See vaadin/vaadin-grid#1197 for reference.

Actually that was accidental. Removed for now, so that the “hover” and “selection” states are applied the same way, only to the data row.

I have considered to keep applying the selection highlight to the detail row too, but decided against that idea. Because the mouse click on the detail row does not actually change selection in the demo, applying selection highlight on the detail row is a bit confusing.

**Comment by jouni** [comment]
:lgtm:

---

Reviewed 11 of 15 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-material-theme/45#-:-LDFonUu5lGIgsIBamn3:bnfp4nl)*

**Pull request merged**